keyvalue/servers/src/main/java/tachyon/worker/keyvalue/KeyValueWorkerClientServiceHandler.java

```diff
@@ -111,4 +115,69 @@ private ByteBuffer getInternal(long blockId, ByteBuffer keyBuffer)
     }
     return null;
   }
+
+  private ByteBufferKeyValuePartitionReader getReader(long sessionId, long lockId, long blockId)
+      throws InvalidWorkerStateException, BlockDoesNotExistException, IOException {
+    BlockReader blockReader = mBlockDataManager.readBlockRemote(sessionId, blockId, lockId);
+    ByteBuffer fileBuffer = blockReader.read(0, blockReader.getLength());
+    ByteBufferKeyValuePartitionReader reader = new ByteBufferKeyValuePartitionReader(fileBuffer);
+    // TODO(binfan): clean fileBuffer which is a direct byte buffer
+    return reader;
+  }
+
+  @Override
+  public List<ByteBuffer> getNextKeys(long blockId, ByteBuffer currentKey, int numKeys)
+      throws TachyonTException, ThriftIOException {
+    try {
+      final long sessionId = Sessions.KEYVALUE_SESSION_ID;
+      final long lockId = mBlockDataManager.lockBlock(sessionId, blockId);
+      try {
```

Comment:
nvm, mBlockDataManager isn't Closeable. Could you use a helper method to reduce the amount of nesting in this method?